packages/hdwallet-core/src/wallet.ts

```diff
@@ -113,11 +114,11 @@ export type Coin = string
  ```
  */
 export function supportsBTC(wallet: any): wallet is BTCWallet {
-  return typeof wallet === 'object' && wallet._supportsBTC === true
+  return isObject(wallet) && (wallet as HDWallet)._supportsBTC === true
```

Comment:
Is there a reason it worked without the type case before? Or is this just an extra safe guard?

Reply:
because `wallet` is `any`, but `isObject` has a return type of `wallet is object`, so at that point TypeScript sees `wallet` as `object`, not as `any`.  `_supportsBTC` is not a property available on `object`.

Reply:
Interesting. Thanks for the explanation!